mysqld-exporter/main.libsonnet

```
@@ -9,11 +11,15 @@ local container = k.core.v1.container;
     user,
     host,
     port=3306,
-    image='prom/mysqld-exporter:v0.13.0',
+    image='prom/mysqld-exporter:v0.16.0',
     tlsMode='preferred',
+    configMapName='%s-cfg' % [name],
   ):: {
     local this = self,
-
+    local versionParts = std.split(std.split(image, ':')[1], '.'),
+    local majorVersion = std.parseInt(std.strReplace(versionParts[0], 'v', '')),
+    local minorVersion = std.parseInt(versionParts[1]),
+    local needsConfigFile = majorVersion > 0 || minorVersion > 14,
```

Nit: Could we add a comment that explains why?

```suggestion
    // Ref: https://github.com/prometheus/mysqld_exporter/releases/tag/v0.15.0
    local needsConfigFile = majorVersion > 0 || minorVersion > 14,
```